### PR TITLE
Document account number verification letter

### PR DESCRIPTION
**Pull request by manuthecoder**
This PR addresses #7488 and provides on-demand verification letter generation through the documents page. 

[Verification Letter for Hack Club HQ.pdf](https://github.com/user-attachments/files/16808396/Verification.Letter.for.Hack.Club.HQ.pdf.pdf)

### CODE REVIEW

File: app/controllers/documents_controller.rb [hunk 1]
```diff
@@ -108,6 +108,20 @@ def fiscal_sponsorship_letter
     end
   end
 
+  def verification_letter
+    authorize @event, policy_class: DocumentPolicy
```
Comment (garyhtou) [inline app/controllers/documents_controller.rb]: We need to handle the case where the Event doesn't have an account number

File: app/views/documents/index.html.erb [hunk 2]
```diff
@@ -28,6 +28,17 @@
...
+    <%= link_to event_verification_letter_path(event_id: @event.slug, format: "pdf") do %>
+      <li class="card card--hover h-100">
+        <div class="overflow-hidden" style="max-height: 250px;">
+          <%= image_tag event_verification_letter_url(event_id: @event.slug, format: "png"), width: 330, style: "max-width: 100%;" %>
+        </div>
+
+        <strong class="h3 block mt1 line-height-2">Verification Letter</strong>
+        <%= user_mention @event.point_of_contact %>
+      </li>
+    <% end %>
```
Comment (garyhtou) [inline app/views/documents/index.html.erb]: don't render if they don't have an account number

File: app/views/documents/verification_letter.pdf.erb [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+    </div>
+
+    <br />
+    <p>
+      Sincerely,
+    </p>
+    <%= image_tag(wicked_pdf_asset_base64('melanie_signature.png'), width: 200, height: 118, dpi: 96) %>
+    <p style="line-height: 1.25">
+      Melanie Smith<br />
+      Fiscal Sponsorship Operations Lead<br />
```
Comment (garyhtou) [inline app/views/documents/verification_letter.pdf.erb]:
```suggestion
      HCB Director of Operations<br />
```

**Review by garyhtou — CHANGES_REQUESTED** [review]
our terminology will need to be slightly different since they don't hold official bank accounts. all bank accounts are owned by Hack Club.

I'll come back later to update the copy

File: app/controllers/documents_controller.rb (hunk 1 above)
Comment (manu-gurudath) [reply app/controllers/documents_controller.rb]: See updated `document_policy.rb`

File: app/controllers/documents_controller.rb [hunk 4]
```diff
@@ -108,6 +108,20 @@ def fiscal_sponsorship_letter
...
+
+    respond_to do |format|
+      format.pdf do
+        render pdf: "verification_letter", page_height: "11in", page_width: "8.5in"
+      end
+
+      format.png do
+        send_data ::DocumentService::VerificationLetter.new(event: @event).run, filename: "verification_letter.png"
```
Comment (garyhtou) [inline app/controllers/documents_controller.rb]:
```suggestion
        send_data ::DocumentService::PreviewVerificationLetter.new(event: @event).run, filename: "verification_letter.png"
```

preview PDF is not loading

File: app/controllers/documents_controller.rb [hunk 5]
```diff
@@ -108,6 +108,20 @@ def fiscal_sponsorship_letter
...
+  def verification_letter
+    authorize @event, policy_class: DocumentPolicy
+
+    respond_to do |format|
+      format.pdf do
+        render pdf: "verification_letter", page_height: "11in", page_width: "8.5in"
+      end
+
+      format.png do
+        send_data ::DocumentService::PreviewVerificationLetter.new(event: @event).run, filename: "verification_letter.png"
```
Comment (garyhtou) [inline app/controllers/documents_controller.rb]: Please include org name in file name

**Review by garyhtou — APPROVED** [review]

**Pull request merged**